### PR TITLE
Deprecate `core.byteLength` and add `await core.info()`

### PR DESCRIPTION
**Pull request by kasperisager**

### CODE REVIEW

**Comment by kasperisager** [comment]
@mafintosh This is ready for review. We should probably bikeshed the format a bit 😄

File: README.md [hunk 1]
```diff
@@ -177,6 +177,18 @@ const updated = await core.update()
...
 
+#### `const stats = await core.stat()`
+
+Get information about this core, such as its total size in bytes.
+
+The object will look like this:
+
+```js
+Stats {
+  size: 742
```
Comment (mafintosh) [inline README.md:188]: what about just saying byteLength here? I think we should add all cheap metadata here also, like the length, fork etc (even tho we have it as props also)

Based on feedback from @andrewosh I think we should name the function info also (or similar if you have ideas)

Comment (mafintosh) [reply README.md:188]: A really cool stat that would help users would be bytesUsed or something also, that infers how much data is actually used if using files (ie fileStat.blocks * 512 vs fileStat.size)

Comment (kasperisager) [reply README.md]: That should really come from the `random-access-*` modules, no?

File: lib/info.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    this.contiguousLength = opts.contiguousLength || 0
+    this.byteLength = opts.byteLength || 0
+    this.padding = opts.padding || 0
+  }
+
+  static empty () {
+    return new Info()
+  }
+
+  static async from (core, padding, snapshot) {
```
Comment (Nazeh) [inline lib/info.js:13]: I am curious why does this need to be async? is it out of anticipation for future information that might need awaiting opening a resource?

Comment (kasperisager) [reply lib/info.js:13]: > is it out of anticipation for future information that might need awaiting opening a resource?

Yep, we already have some ideas for additional information that will definitely result in IO.

**Comment by kasperisager** [comment]
@mafintosh This is ready for another review.

**Comment by mafintosh** [comment]
Love it

**Pull request merged**